.github/workflows/test_and_deploy.yml

```diff
@@ -0,0 +1,137 @@
+---
+name: "CI & CD"
+
+on:
+  workflow_dispatch:
+  pull_request:
+  release:
+    types:
+    - published
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
+  cancel-in-progress: true
+
+env:
+  CIBW_TEST_REQUIRES: pytest
+  CIBW_TEST_COMMAND: "pytest {project} --ignore={project}//tests//structure//test_trajectory.py --ignore={project}//tests//sequence//align//test_statistics.py --ignore={project}//tests//application --ignore={project}//tests//database  --ignore={project}//tests//test_doctest.py --ignore={project}//tests//test_modname.py --durations=50"
+  CIBW_DEPENDENCY_VERSIONS: "pinned"
+  # Once GHA and cibuildwheel are updated this can be removed
+  # mussllinux takes 6+ hrs to build and test so ignore it
+  CIBW_TEST_SKIP: "*musllinux* *-macosx_arm64"
+
+jobs:
+  generate-wheels-matrix:
+    name: "Generate wheels matrix"
+    runs-on: "ubuntu-latest"
+    outputs:
+      include: ${{ steps.set-matrix.outputs.include }}
+    steps:
+      - uses: actions/checkout@v3
+      - name: Install cibuildwheel
+        # MAKE SURE THIS STAYS IN SYNC WITH THE LOWER GHA cibuildwheel
+        run: pipx install cibuildwheel==2.11.4
+      - id: set-matrix
+        run: |
+          MATRIX=$(
+            {
+              cibuildwheel --print-build-identifiers --platform linux \
+              | jq -nRc '{"only": inputs, "os": "ubuntu-20.04"}' \
+              && cibuildwheel --print-build-identifiers --platform macos \
+              | jq -nRc '{"only": inputs, "os": "macos-11"}' \
+              && cibuildwheel --print-build-identifiers --platform windows \
+              | jq -nRc '{"only": inputs, "os": "windows-2019"}'
+            } | jq -sc
+          )
+          echo "include=$MATRIX" | tee -a $GITHUB_OUTPUT
+        env:
+          CIBW_ARCHS_LINUX: "x86_64 aarch64"
+          CIBW_ARCHS_MACOS: "x86_64 arm64"
+          CIBW_ARCHS_WINDOWS: "x86 AMD64"
+          # Skip musllinux because it takes too long to compile on GHA
+          # since it is emulated. (6+ hours)
+          # *note* most of the build time is actually numpy for musllinux
+          CIBW_SKIP: "*musllinux* *-manylinux_i686 *-musllinux_i686 *-win32 pp*"
+
+  test-biotite-and-build-wheels:
+    name: "Build & Test"
+    needs: "generate-wheels-matrix"
+    strategy:
+      matrix:
+        include: ${{ fromJson(needs.generate-wheels-matrix.outputs.include) }}
+    runs-on: ${{ matrix.os }}
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Set up QEMU
+        if: runner.os == 'Linux'
+        uses: docker/setup-qemu-action@v2
+        with:
+          platforms: all
+
+      - name: Build & (optionally) Test wheels
+        # MAKE SURE THIS STAYS IN SYNC WITH THE UPPER pipx call to cibuildwheel
+        uses: pypa/cibuildwheel@v2.11.4
+        with:
+          only: ${{ matrix.only }}
```

Comment:
yes it is slightly different.
https://github.com/pypa/cibuildwheel/blob/main/action.yml

Reply:
I wonder if this difference matter in this case, especially as the *cibuildwheel* docs also give usage examples without the action. Along with the lower complexity this would resolve the issue with synced version number, which gives me a little bit of a headache.